packages/analytics-core/src/plugins/destination.ts

```diff
@@ -244,10 +244,7 @@ export class Destination implements DestinationPlugin {
    * 2) response comes back for a request
    */
   saveEvents() {
-    if (!this.config.saveEvents) {
-      return;
-    }
     const events = Array.from(this.queue.map((context) => context.event));
-    void this.config.storageProvider.set(this.storageKey, events);
+    void this.config.storageProvider?.set(this.storageKey, events);
```

Comment:
This part works fine. I am thinking maybe we can check `this.config.storageProvider` first and if there is no storage provider, we can skip iterating the events.

Reply:
That's better, I will make the change.

Reply:
Updated, @dnshi